src/rust/engine/src/tasks.rs

```diff
@@ -130,6 +130,10 @@ impl Tasks {
 
   pub fn intrinsics_set(&mut self, types: &Types) {
     let intrinsics = vec![
+      Intrinsic {
+        product: types.directory_digest,
+        input: types.input_file_content,
```

Comment:
I like the idea of using `FilesContent` for multiple files by default! And a path ordering constraint sounds like exactly the right idea (I believe we'd probably have to do this sorting within the `FilesContent` constructor in order for the engine to cache it?). I don't believe this would be too difficult to implement.